interpreter/cling/lib/Interpreter/ASTImportSource.cpp

```diff
@@ -168,4 +201,55 @@ namespace cling {
       }
       return false;
     }
+
+    ///\brief Make available to child all decls in parent's decl context
+    /// that corresponds to child decl context.
+    void ASTImportSource::completeVisibleDeclsMap(
+                                  const clang::DeclContext *childDeclContext) {
+      assert (childDeclContext && "No child decl context!");
+
+      if (!childDeclContext->hasExternalVisibleStorage())
+        return;
+
+      // Search in the map of the stored Decl Contexts for this
+      // Decl Context.
+      std::map<const clang::DeclContext *, clang::DeclContext *>::iterator I;
+      // If childCurrentDeclContext was found before and is already in the map,
+      // then do the lookup using the stored pointer.
+      if ((I = m_DeclContexts_map.find(childDeclContext))
+          != m_DeclContexts_map.end()) {
+
+        DeclContext *parentDeclContext = I->second;
+
+        Decl *fromDeclContext = Decl::castFromDeclContext(parentDeclContext);
+        ASTContext &from_ASTContext = fromDeclContext->getASTContext();
+
+        Decl *toDeclContext = Decl::castFromDeclContext(childDeclContext);
+        ASTContext &to_ASTContext = toDeclContext->getASTContext();
+
+        // Prepare to import the Decl(Context)  we found in the
+        // child interpreter by getting the file managers from
+        // each interpreter.
+        FileManager &child_FM = m_child_Interp->getCI()->getFileManager();
+        FileManager &parent_FM = m_parent_Interp->getCI()->getFileManager();
+
+        // Cling's ASTImporter
+        ClingASTImporter importer(to_ASTContext, child_FM,
+                             from_ASTContext, parent_FM,
+                             /*MinimalImport : ON*/ true, *this);
+
+        for (DeclContext::decl_iterator I_decl = parentDeclContext->decls_begin(),
+          E_decl = parentDeclContext->decls_end(); I_decl != E_decl; ++I_decl) {
+          if (NamedDecl* parentNamedDecl = llvm::dyn_cast<NamedDecl>(*I_decl)) {
+            DeclarationName newDeclName = parentNamedDecl->getDeclName();
+            ImportDecl(parentNamedDecl, importer, newDeclName, newDeclName,
+                       childDeclContext);
+          }
+        }
+
+        const_cast<DeclContext *>(childDeclContext)->
```

Comment:
...and the answer is "yes": `ASTReader::completeVisibleDeclsMap(const DeclContext *DC)` has
```
  const_cast<DeclContext *>(DC)->setHasExternalVisibleStorage(false);
```

Fair enough...